### ISSUE TITLE
Incorrect assumption about FCM endpoint

### ISSUE THREAD

**Issue by arjan** [tuvistavie/elixir-web-push-encryption#14]
[This code][link] asserts that an auth key (GCM sender ID) is required when a push subscription is sent from FCM. However, this is not true because FCM fully implements VAPID, as can be read [here][link2]: "With VAPID you no longer need to sign up for an account with GCM".

My suggestion therefore is to remove this function clause altogether, as well as [this part](https://github.com/danhper/elixir-web-push-encryption/blob/master/lib/web_push_encryption/push.ex#L69-L70).

[link]: https://github.com/danhper/elixir-web-push-encryption/blob/master/lib/web_push_encryption/push.ex#L33

[link2]: https://developers.google.com/web/updates/2016/07/web-push-interop-wins#introducing_vapid_for_server_identification

**Comment by danhper** [comment]
Hi, thanks for reporting the issue.
I suppose that was the situation when I first implemented this library but if it not required anymore this should indeed be fixed.
It would be great if you had the time to send a PR with the changes you suggested.
Thanks a lot.

**Comment by arjan** [comment]
Yes, only, this will effectively reverse #13 so I am curious why @splagemann fixed this in the first place...

**Comment by splagemann** [comment]
I'm pretty sure that I had issues without an auth key. But I can check that again and I'm happy to have that removed.

**Comment by arjan** [comment]
I had issues *with* an auth key :-) 
Anyway I think it's possible to support both code paths. Could you check whether the auth key path works for you?

**Comment by drobban** [comment]
What is the status of this issue?
